app/backend/src/tests/test_bg_jobs.py

```diff
@@ -649,6 +652,88 @@ def test_process_send_onboarding_emails(db):
+    user2, token2 = generate_user(email="user2@couchers.org.invalid", name="User 2")
+
+    # case 2: host left ref (surfer needs an email)
+    # host
+    user3, token3 = generate_user(email="user3@couchers.org.invalid", name="User 3")
+    # surfer
+    user4, token4 = generate_user(email="user4@couchers.org.invalid", name="User 4")
+
+    # case 3: surfer left ref (host needs an email)
+    # host
+    user5, token5 = generate_user(email="user5@couchers.org.invalid", name="User 5")
+    # surfer
+    user6, token6 = generate_user(email="user6@couchers.org.invalid", name="User 6")
+
+    # case 4: neither left ref (host & surfer need an email)
+    # host
+    user7, token7 = generate_user(email="user7@couchers.org.invalid", name="User 7")
+    # surfer
+    user8, token8 = generate_user(email="user8@couchers.org.invalid", name="User 8")
+
+    # case 5: neither left ref, but host blocked surfer, so neither should get an email
+    # host
+    user9, token9 = generate_user(email="user9@couchers.org.invalid", name="User 9")
+    # surfer
+    user10, token10 = generate_user(email="user10@couchers.org.invalid", name="User 10")
+
+    make_user_block(user9, user10)
+
+    with session_scope() as session:
+        # note host requests have one day longer age
+
+        # case 1: bidirectional (no emails)
+        ref1, hr1 = create_host_reference(session, user2.id, user1.id, timedelta(days=7), surfing=True)
+        create_host_reference(session, user1.id, user2.id, timedelta(days=7), host_request_id=hr1)
+
+        # case 2: host left ref (surfer needs an email)
+        ref2, hr2 = create_host_reference(session, user3.id, user4.id, timedelta(days=11), surfing=False)
+
+        # case 3: surfer left ref (host needs an email)
+        ref3, hr3 = create_host_reference(session, user6.id, user5.id, timedelta(days=9), surfing=True)
+
+        # case 4: neither left ref (host & surfer need an email)
+        hr4 = create_host_request(session, user7.id, user8.id, timedelta(days=4))
+
+        # case 5: neither left ref, but host blocked surfer, so neither should get an email
+        hr5 = create_host_request(session, user9.id, user10.id, timedelta(days=7))
+
+    expected_emails = [
+        ("user4@couchers.org.invalid", "You have 3 days to write a reference for User 3!"),
+        ("user5@couchers.org.invalid", "You have 7 days to write a reference for User 6!"),
+        ("user7@couchers.org.invalid", "You have 12 days to write a reference for User 8!"),
+        ("user8@couchers.org.invalid", "You have 12 days to write a reference for User 7!"),
+    ]
+
+    process_send_reference_reminders(empty_pb2.Empty())
+
+    while process_job():
+        pass
+
+    with session_scope() as session:
+        emails = [
+            (email.recipient, email.subject)
+            for email in session.execute(select(Email).order_by(Email.recipient.asc())).scalars().all()
+        ]
+
+        print(emails)
+        print(expected_emails)
```

Comment:
Leftover prints

Reply:
I left them on purpose this time, it's useful for debugging.